lib/caasp.pm

```diff
@@ -264,4 +268,44 @@ sub script_retry {
     }
 }
 
+# All events ordered by execution
+my %events = (
+    support_server_ready     => 'Wait for dhcp, dns, ntp, ..',
+    VELUM_STARTED            => 'Wait until velum starts to login there from controller',
+    VELUM_CONFIGURED         => 'Velum has to be configured before autoyast installations start',
+    NODES_ACCEPTED           => 'Wait until salt-keys are accepted to set password on autoyast nodes',
+    REBOOT_FINISHED          => 'Re-login when system rebooted after update/reboot test',
+    DELAYED_WORKER_INSTALLED => 'Wait until we have new node that we can add to existing cluster',
+    DELAYED_NODES_ACCEPTED   => 'Wait until salt-keys are accepted to set password on autoyast nodes',
+    CNTRL_FINISHED           => 'Wait on CaaSP nodes until controller finishes testing',
+);
+
+# CaaSP specific unpausing
+sub unpause {
+    my $event = shift;
+
+    # Handle cluster failure on controller node
+    if (uc($event) eq 'ALL') {
+        lockapi::unpause keys %events;
+    }
+    else {
+        lockapi::unpause $event;
+    }
+}
+
+# CaaSP specific pausing
+sub pause_until {
```

Comment:
IMHO a more common word in this context is "wait", which could be used like `wait_for`

Reply:
I was thinking about this - I liked wait_for, but I could nod find good `unwait` word - I might rename it later...will see